scripts/install/requestrr.sh

```diff
@@ -0,0 +1,178 @@
+. /etc/swizzin/sources/functions/os
+
+echo_progress_start "Downloading source files"
+case "$(_os_arch)" in
+    "amd64") dlurl=$(curl -sNL https://api.github.com/repos/darkalfx/requestrr/releases/latest | grep -Po 'ht(.*)linux-x64(.*)zip') >> ${log} 2>&1 ;;
+    "armhf") dlurl=$(curl -sNL https://api.github.com/repos/darkalfx/requestrr/releases/latest | grep -Po 'ht(.*)linux-arm(.*)zip') >> ${log} 2>&1 ;;
+    "arm64") dlurl=$(curl -sNL https://api.github.com/repos/darkalfx/requestrr/releases/latest | grep -Po 'ht(.*)linux-arm64(.*)zip') >> ${log} 2>&1 ;;
+    *)
+        echo_error "Arch not supported"
+        exit 1
+        ;;
+esac
+
+if ! curl "$dlurl" -L -o /tmp/requestrr.zip >> "$log" 2>&1; then
+    echo_error "Download failed, exiting"
+    exit 1
+fi
+echo_progress_done "Source downloaded"
+
+echo_progress_start "Extracting archive"
+unzip -q /tmp/requestrr.zip -d /opt/ >> "$log" 2>&1
+rm /tmp/requestrr.zip
+mv /opt/requestrr* /opt/requestrr
+echo_progress_done "Archive extracted"
+
+touch /install/.requestrr.lock
+
+echo_progress_start "Creating requestrr user and setting permssions"
+useradd --system -d /opt/requestrr/ requestrr
+chown -R requestrr:requestrr /opt/requestrr/
+chmod +x /opt/requestrr/Requestrr.WebApi
+echo_progress_done "Requestrr user has been created & permissions set."
+
+echo_progress_start "Applying Requestrr config..."
+cat > /opt/requestrr/SettingsTemplate.json << CFG
+{
+  "Authentication": {
+    "Username": "",
+    "Password": "",
+    "PrivateKey": "[PRIVATEKEY]"
+  },
+  "ChatClients": {
+    "Discord": {
+      "BotToken": "",
+      "ClientId": "",
+      "StatusMessage": "!help",
+      "TvShowRoles": [],
+      "MovieRoles": [],
+      "MonitoredChannels": [],
+      "EnableRequestsThroughDirectMessages": false,
+      "AutomaticallyNotifyRequesters": true,
+      "NotificationMode": "PrivateMessages",
+      "NotificationChannels": [],
+      "AutomaticallyPurgeCommandMessages": false,
+      "DisplayHelpCommandInDMs": true
+    }
+  },
+  "DownloadClients": {
+    "Ombi": {
+      "Hostname": "",
+      "Port": 3579,
+      "ApiKey": "",
+      "ApiUsername": "",
+      "BaseUrl": "",
+      "UseSSL": false,
+      "Version": "3"
+    },
+    "Overseerr": {
+      "Hostname": "",
+      "Port": 5055,
+      "ApiKey": "",
+      "DefaultApiUserID": "",
+      "UseSSL": false,
+      "Version": "1"
+    },
+    "Radarr": {
+      "Hostname": "",
+      "Port": 7878,
+      "ApiKey": "",
+      "BaseUrl": "",
+      "MovieProfileId": "1",
+      "MovieRootFolder": "",
+      "MovieMinimumAvailability": "",
+      "MovieTags": [],
+      "AnimeProfileId": "1",
+      "AnimeRootFolder": "",
+      "AnimeMinimumAvailability": "",
+      "AnimeTags": [],
+      "SearchNewRequests": true,
+      "MonitorNewRequests": true,
+      "UseSSL": false,
+      "Version": "2"
+    },
+    "Sonarr": {
+      "Hostname": "",
+      "Port": 8989,
+      "ApiKey": "",
+      "BaseUrl": "",
```

Comment:
we _could_ fill these in, seeing as they are all in the config xml file in ~/.config. Would be slick. Could help out with that.

Reply:
I wasn’t sure how this password was being salted and hashed. It didn’t look like htpasswd; but I’ll go for a look see and do some playing with it this afternoon.

Reply:
Going to def. play with this in a bit -- going for a walk, then will get started on it.

Reply:
Have consulted with a few peeps and they made the argument that this should be opt-in -- what says you?

Reply:
Nah make it automatic if it's installed, whoever would want to opt out will know what they need to do. They'll have to change things anyway. 80% usecase says opt-in.

Reply:
@flying-sausages, what if I do an {ask} y/n?; or would this screw up on an unattended install...

/b

Reply:
nah let's just do it default lol. opt-out through environment variable _maybe_. Bazarr does the same thing. Ombi should too, and we should do the same thing with overseerr whenever it's possible.

Reply:
Alrighty. I'll merge in the stuffs and do some testing 😜